sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamEnumerableConverter.java

```diff
@@ -238,16 +238,29 @@ public void startBundle(StartBundleContext context) {
 
     @ProcessElement
     public void processElement(ProcessContext context) {
-      Object[] avaticaRow = rowToAvatica(context.element());
-      if (avaticaRow.length == 1) {
-        values.add(avaticaRow[0]);
+      values.add(rowToObjectArray(context.element()));
+    }
+  }
+
+  private static List<Object> objectArrayToAvatica(Queue<Object[]> values) {
+    List<Object> ret = new ArrayList<>();
+    Iterator<Object[]> iterator = values.iterator();
+    while (iterator.hasNext()) {
+      Object[] objects = iterator.next();
+      if (objects.length == 1) {
+        // if objects.length == 1, that means input Row contains only 1 column/element,
+        // then an Object instead of Object[] should be returned because of CalciteResultSet's
+        // behaviour that tries to convert one column row to an Object.
+        ret.add(objects[0]);
       } else {
-        values.add(avaticaRow);
+        ret.add(objects);
       }
     }
+
+    return ret;
   }
 
-  private static Object[] rowToAvatica(Row row) {
+  private static Object[] rowToObjectArray(Row row) {
```

Comment:
I think the name `rowToAvatica` is still appropriate here.

Reply:
Ok, I changed the name back.